include/boost/fixed_string/fixed_string.hpp

```diff
@@ -2599,9 +2599,52 @@ fixed_string(CharT(&)[N]) ->
     fixed_string<N, CharT, std::char_traits<CharT>>;
 #endif
 
+//------------------------------------------------------------------------------
+//
+// Hashing
+//
+//------------------------------------------------------------------------------
+
+#ifndef BOOST_FIXED_STRING_STANDALONE
+// hash_value overload for Boost.Container_Hash
+template <std::size_t N, 
+    typename CharT, 
+    typename Traits>
+std::size_t 
+hash_value(
+  const fixed_string<N, CharT, Traits>& str)
+{
+#ifndef BOOST_FIXED_STRING_STANDALONE
+  return boost::hash_range(str.begin(), str.end());
+#else

```

Comment:
The outer `#ifndef` doesn't make sense if you have the inner one. i.e. How would it ever hit the inner `#else`?